frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/services/MfaResetIPVAuthorizationService.java

```diff
@@ -0,0 +1,200 @@
+        EncryptedJWT requestJWT =
+                constructMfaResetAuthorizationJWT(state, subject, claims, clientSessionId);
+
+        AuthorizationRequest.Builder authRequestBuilder =
+                new AuthorizationRequest.Builder(
+                                new ResponseType(ResponseType.Value.CODE),
+                                new ClientID(configurationService.getIPVAuthorisationClientId()))
+                        .endpointURI(configurationService.getIPVAuthorisationURI())
+                        .requestObject(requestJWT);
+
+        AuthorizationRequest ipvAuthorisationRequest = authRequestBuilder.build();
+        String ipvAuthorisationRequestURI = ipvAuthorisationRequest.toURI().toString();
+        LOG.info("MFA reset JAR created, redirect URI {}", ipvAuthorisationRequestURI);
+
+        storeState(session.getSessionId(), state);
+
+        auditService.submitAuditEvent(REVERIFY_AUTHORISATION_REQUESTED, auditContext);
+
+        cloudwatchMetricsService.incrementMfaResetHandoffCount();
+
+        return generateApiGatewayProxyResponse(
+                302, "", Map.of(ResponseHeaders.LOCATION, ipvAuthorisationRequestURI), null);
+    }
+
+    private EncryptedJWT constructMfaResetAuthorizationJWT(
+            State state, Subject subject, ClaimsSetRequest claims, String clientSessionId) {
+        LOG.info("Generating MFA Reset request JWT");
+        JWTClaimsSet mfaResetAuthorizationClaims =
+                createMfaResetAuthorizationClaims(state, subject, claims, clientSessionId);
+        SignedJWT signedJWT =
+                jwtService.signJWT(
+                        SIGNING_ALGORITHM,
+                        mfaResetAuthorizationClaims,
+                        configurationService.getMfaResetJarSigningKeyAlias());
+        LOG.info("Created Signed MFA Reset JWT");
+
+        EncryptedJWT encryptedJWT = jwtService.encryptJWT(signedJWT, getPublicKey());
+        LOG.info("Created encrypted MFA Reset request JWT");
+        return encryptedJWT;
+    }
+
+    private JWTClaimsSet createMfaResetAuthorizationClaims(
+            State state, Subject subject, ClaimsSetRequest claims, String clientSessionId) {
+        LOG.info("Creating MFA Reset Authorization claims");
+        String jwtID = IdGenerator.generate();
+        Date expiryDate = nowClock.nowPlus(3, ChronoUnit.MINUTES);
+        OIDCClaimsRequest claimsRequest = new OIDCClaimsRequest().withUserInfoClaimsRequest(claims);
+        JWTClaimsSet.Builder claimsBuilder =
+                new JWTClaimsSet.Builder()
+                        .issuer(configurationService.getAuthIssuerClaim())
+                        .audience(configurationService.getIPVAudience())
+                        .expirationTime(expiryDate)
+                        .subject(subject.getValue())
+                        .issueTime(nowClock.now())
+                        .jwtID(jwtID)
+                        .notBeforeTime(nowClock.now())
```

Comment:
Probably not an issue, but this opens us up to the possibility that the `iat` and the `nbf` don't match exactly, and that the token lifetime is slightly less than 3 mins

Reply:
I've tried to fix this by taking a timestamp for the iat and then taking the expiry time immediately after, hopefully this minimises the gap